Mathlib/Algebra/Vertex/VertexOperator.lean

```diff
@@ -0,0 +1,99 @@
+/-
+Copyright (c) 2024 Scott Carnahan. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Scott Carnahan
+-/
+import Mathlib.Algebra.Vertex.HVertexOperator
+import Mathlib.Data.Int.Interval
+
+/-!
+# Vertex operators
+In this file we introduce vertex operators as linear maps to Laurent series.
+
+## Definitions
+ * `VertexOperator` is an `R`-linear map from an `R`-module `V` to `LaurentSeries V`.
+ * `VertexOperator.ncoef` is the coefficient of a vertex operator under normalized indexing.
+
+## TODO
+ * `HasseDerivative` : A divided-power derivative.
+ * `Locality` : A weak form of commutativity.
+ * `Residue products` : A family of products on `VertexOperator R V` parametrized by integers.
+
+## References
+ * G. Mason `Vertex rings and Pierce bundles` ArXiv 1707.00328
+ * A. Matsuo, K. Nagatomo `On axioms for a vertex algebra and locality of quantum fields`
+  arXiv:hep-th/9706118
+-/
+
+noncomputable section
+
+variable {R V : Type*} [CommRing R] [AddCommGroup V] [Module R V]
+
+/-- A vertex operator over a commutative ring `R` is an `R`-linear map from an `R`-module `V` to
+Laurent series with coefficients in `V`.  We write this as a specialization of the heterogeneous
+case. -/
+abbrev VertexOperator (R : Type*) (V : Type*) [CommRing R] [AddCommGroup V]
+    [Module R V] := HVertexOperator ℤ R V V
+
+namespace VertexOperator
+
+open HVertexOperator
+
+@[ext]
+theorem ext (A B : VertexOperator R V) (h : ∀ v : V, A v = B v) :
+    A = B := LinearMap.ext h
+
+/-- The coefficient of a vertex operator under normalized indexing. -/
+def ncoef {R} [CommRing R] [AddCommGroup V] [Module R V] (A : VertexOperator R V) (n : ℤ) :
```

Comment:
Could this be `ncoeff`, for consistency with our abbreviations of "coefficient" elsewhere?